app/views/dialogue.py

```diff
@@ -279,31 +280,216 @@ def show_fatal_error(
+        :rtype: int
+        """
+        return self.exec()
+
+
+class _BaseMessageBox(QMessageBox):
+    """Base message box class for all custom message boxes."""
+
+    _dialogue_type = "base message box"
+
+    def __init__(
+        self,
+        title: str,
+        text: str,
+        information: str,
+        icon: QMessageBox.Icon,
+        details: str | None = None,
+        text_format: Qt.TextFormat = Qt.TextFormat.RichText,
+        modal: bool = True,
+        parent: QWidget | None = None,
+    ):
+        super().__init__(parent)
+        self.setWindowTitle(title)
+        self.setModal(modal)
         self.setObjectName("dialogue")
+        self.setTextFormat(text_format)
+        self.setIcon(icon)
+        # Set text to be bold via rich text
+        if text_format == Qt.TextFormat.RichText:
+            text = f"<b>{text}</b>"
+        self.setText(text)
+        self.setInformativeText(information)
+        if details is not None:
+            self.setDetailedText(details)
 
         # Dynamic sizing
         self.setSizePolicy(
             QSizePolicy(QSizePolicy.Policy.Expanding, QSizePolicy.Policy.Expanding)
         )
 
+    def exec(self) -> int:
+        """Executes the message box and returns the result.
+
+        :return: The result of the message box
+        :rtype: int
+        """
+        logger.info(
+            f"Showing {self._dialogue_type} with title: [{self.windowTitle()}], text: [{self.text()}], information: [{self.informativeText()}], details: [{self.detailedText()}]"
+        )
+        result = super().exec()
+        logger.info(
+            f"Finished showing {self._dialogue_type} [{self.windowTitle()}] with result: {result}"
+        )
+        return result
+
+    def exec_(self) -> int:
+        """Executes the message box and returns the result.
+
+        :return: The result of the message box
+        :rtype: int
+        """
+        return self.exec()
+
+
+class BinaryChoiceDialog(_BaseMessageBox):
+    """Custom message box to display a binary choice message box."""
+
+    def __init__(
+        self,
+        title: str = "",
+        text: str = "",
+        information: str = "",
+        details: str | None = None,
+        positive_text: str | None = None,
+        negative_text: str | None = None,
+        positive_btn: QMessageBox.StandardButton = QMessageBox.StandardButton.Yes,
+        negative_btn: QMessageBox.StandardButton = QMessageBox.StandardButton.Cancel,
+        default_negative: bool = True,
+        icon: QMessageBox.Icon = QMessageBox.Icon.Question,
+        modal: bool = True,
+        parent: QWidget | None = None,
+    ) -> None:
+        """Initializes the binary choice dialog.
+        Used to display a binary choice message box.
+        Always has two buttons, one positive and one negative.
+        These buttons cannot be the same type.
+
+        :param title: The title of the message box
+        :type title: str, optional
```

Comment:
Curious if the added verboseness of having types in the docstrings are needed, seeing as we already have type annotations. For functions like these it seems like a lot

Reply:
The types in the doc string are a part of the sphinx standard which is what we currently are following (at least according to the wiki). It's probably required for the auto-generated html documentation that sphinx would generate, though we aren't using that right now.